app/src/views/private/components/drawer-collection.vue

```diff
@@ -16,15 +18,20 @@
 			:title="t('select_item')"
 			:small-header="currentLayout?.smallHeader"
 			:header-shadow="currentLayout?.headerShadow"
+			v-bind="drawerProps"
 			@cancel="cancel"
 		>
+			<template v-for="(_, slot) of $slots" #[slot]="scope">
+				<slot :name="slot" v-bind="scope" />
+			</template>
+
 			<template #subtitle>
-				<v-breadcrumb :items="[{ name: collectionInfo.name, disabled: true }]" />
+				<v-breadcrumb :items="[{ name: collectionInfo?.name, disabled: true }]" />
 			</template>
 
 			<template #title-outer:prepend>
 				<v-button class="header-icon" rounded icon secondary disabled>
-					<v-icon :name="collectionInfo.icon" :color="collectionInfo.color" />
+					<v-icon :name="collectionInfo?.icon" :color="collectionInfo?.color" />
```

Comment:
```js
<v-icon :name="collectionInfo?.icon" :color="collectionInfo?.color" />
```
if `collectionInfo` is undefined here should we not hide the icon component altogether? 🤔

Reply:
If collectionInfo is undefined, this whole page shouldn't render (it should be a 404) 🤔

Reply:
Should the type not reflect that then 🤔 currently it can be null all over this component according to the typing

Reply:
Yup! Welcome to the TypeScript Modernization Program for the Application, we're happy to have you. Keep in mind that Vue didn't really support TypeScript back when we started it, so a lot of these types have been added in over time. This is one of many rough edges that we'll be cleaning up over time, starting with the script[setup] business 🙂